bql/planner/data_access.go

```diff
@@ -409,62 +409,80 @@ func simpleFetch(ctx context.Context, gs []storage.Graph, cls *semantic.GraphCla
 // table. The semantic graph clause is also passed to be able to identify what
 // bindings to set.
 func addTriples(ts <-chan *triple.Triple, cls *semantic.GraphClause, tbl *table.Table) error {
-	for t := range ts {
+	shouldIgnoreTriple := func(t *triple.Triple) (bool, error) {
```

Comment:
nit: Although this function is only used in the context of addTriples, it is hindering readability (breaking the flow of "adding Triple logic)/ Let's move it to the package level. This also helps to make this function pure (in the sense that it won't be able to depend on variables from the outter scope) wdyt?

Reply:
When writing `shouldIgnoreTriple` I thought about it but also thought that the argument of keeping this function local (as it is used only by `addTriples`) was stronger.
Thank you once again for the input Jose, I am new to Go so I appreaciate a lot all these hints. I do agree that with `shouldIgnoreTriple` in the package level we have a much more readable `addTriples`. 👍 
Then, I have just done as you suggested. 😀